src/ui/EastRegion.js

```diff
@@ -197,12 +214,13 @@ EastRegion = function(c) {
                 fieldLabel: i18n.sharing,
                 labelStyle: 'padding-top:0',
                 style: 'margin-bottom:3px',
-                value: getSharingText(layout) + '<span style="padding-left:10px">' + getLink(editText, false, true) + '</span>',
+                value: getSharingText(layout) + (layout && layout.permission === 'write' ? 
+                    '<span style="padding-left:10px">' + getLink(editText, false, true) + '</span>' : ''),
                 cls: 'interpretationDetailsField',
                 listeners: {
                     'render': function(label) {
                         label.getEl().on('click', function() {
-                            instanceManager.getSharingById(instanceManager.getStateFavoriteId(), function(r)  {
+                            instanceManager.getSharingById(instanceManager.getStateFavoriteId(), function(r) {
```

Comment:
You have removed Edit link but the text 'Public: None' is still a link. Can you make remove the link for the text as well when users don't have the right permissions? Probably you didn't realize because the text doesn't look like a link but actually it is a link